Cm trt1

eyalpe opened the pull request.

Commit:
CMakeLists: Some improvement and indentation fixes

Mainly, modified tensorflow subdirectory, use a subdirectory relative to
project path for accessing tensorflow .so and .h on linux-pc build.

Commit:
Removed unused method GetResultScores from mbInterfaceCM

Commit:
testStressSmall: Minor change to avoid compilation warnings

Commit:
Make an abstract base class for color-model interface

In order to allow easy addition of alternative implementation.

Commit:
Modified InterfaceCM class to use InterfaceCMbace

Use newly introduced interfaceCMbace abstract class as a base class for
InterfaceCM without functionallity changes.

Commit:
Remove "/home/borisef/..." prefix-paths in configuration files

Use current-working-directory as prePath instead of Boris's home directory

Commit:
Fix compilation warnings

Commit:
InterfaceCMTtrt: Added an implementation for loading tensotr-rt engines

Check filetype field in the configuration file to decide which cm to
instantiate. If filetype is ".engine", we will use an InterfaceCMTtrt
instance, if it is ".pb" we will use the old InterfaceCM (tensor-flow),
otherwise we fail.

Commit:
Modify CMakeLists.txt to support compilation with tensorrt

Commit:
Some fixes to allow arm64 (Jetson) compilation

Commit:
modified color names mapping

Commit:
color-model: read height, width, max_batch from config

Commit:
temp: debug prints

Commit:
WIP: color model - batch run

Commit:
color model: trt engine support batches. TODO: clean debug prints.

Commit:
color-model: Clean debug prints

Commit:
Fix compilation errors/warnings in TEST_MODE, after re-base.

Commit:
Support compilation with and without Tensor-RT

borisef commented on include/cppflowCM/InterfaceCM.h:
```diff
@@ -1,47 +1,38 @@
+  // bool m_hardBatchSize = false;
+  //float m_tileMargin = 0.2;
 
-#ifdef TEST_MODE
```
I need this test mode

eyalpe replied:
resolved, but in 'src/Object_Detection_Handler.cpp'.

eyalpe commented on include/cppflowCM/InterfaceCMbase.h:
```diff
@@ -0,0 +1,128 @@
+    }
+
+#ifdef TEST_MODE
+    // map color id to color-name
```
take color name from odutils

borisef commented on CMakeLists.txt:
```diff
@@ -11,7 +11,7 @@ message($ENV{USER})
 
 
-set(TEST_MODE TRUE)
+#set(TEST_MODE TRUE)
```
make test_mode FALSE instead

eyalpe replied:
done

borisef commented on CMakeLists.txt:
```diff
@@ -63,8 +63,9 @@ if(OS_WINDOWS)
     include_directories(PRIVATE include)
-    find_package( OpenCV REQUIRED)
-    
+    set(OpenCV_DIR /usr/lib/x86_64-linux-gnu/cmake/opencv4)
```
avoid x86_64

Commit:
Force batch size to 1 on test mode

Commit:
Explicitly set TEST_MODE to FALSE

Instead of using a comment which keeps it undefined.

Commit:
od-utils/GetColorString: Avoid re-definitions of color id-to-name mapping

Modified GetColorString() to use an existing map (mapOfcolors) that maps
color codes (OD::e_OD_TargetColor) to color names, instead of implementing
a hard coded mapping in the function. This way avoid the possibility of
contradictions in color id to name mappings.

Commit:
CM-TRT (TEST-MODE): Take color name from od-utils' GetColorString()

Instead of implementing additional mapping which is error-prone tactic.

Commit:
CMakeLists.txt: Use ${CMAKE_SYSTEM_PROCESSOR} instead of hard-coded arch

borisef committed:
Merge branch 'master' into cm_trt1
